[FIX UNIT TEST] Mock ip_addrs() in utils/minions.py unit test

agraul opened the pull request:
### What does this PR do?

Previously the test used `salt.utils.network.ip_addrs()` in the same way that the tested code did. This worked well as long as at least one IP address was returned by `salt.utils.network.ip_addrs()`.

Since this is a unit test, it should not depend on the environment, it should just work™, even if there are no real IP addresses assigned to the system (or container) that runs the test.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/60345 (should I create one as well or is the description above enough?) Yes! Feel free to use the `Test Failure` template I have made one for you, here.

### Previous Behavior
When `salt.utils.network.ip_addrs()` returned no IPs, the test failed.

### New Behavior
`salt.utils.network.ip_addrs()` is mocked and always returns IPs. The test can't fail in this way anymore.

### Merge requirements satisfied?
- [x] Tests written/updated

### Commits signed with GPG?
Yes

waynew commented on tests/pytests/unit/utils/test_minions.py:
```diff
@@ -8,15 +8,16 @@ def test_connected_ids():
-    ip = salt.utils.network.ip_addrs()
-    mdata = {"grains": {"ipv4": ip, "ipv6": []}}
-    ckminions = salt.utils.minions.CkMinions({"minion_data_cache": True})
+    ips = {"192.168.1.1", "10.10.10.10"}
```
**suggestion** use 203.0.113.0/24 or other TEST-NET addresses (https://en.m.wikipedia.org/wiki/Reserved_IP_addresses) to explicitly communicate these are not real IP addresses

agraul replied:
I didn't know about that there were reserved IP addresses that can be used for this purpose, that's useful. I'll change the IPs. Thanks!

agraul committed:
Mock ip_addrs() in utils/minions.py unit test

Previously the test used `salt.utils.network.ip_addrs()' in the same way
that the tested code did. This worked well as long as at least one IP
address was returned by `salt.utils.network.ip_addrs()'.

Since this is a unit test, it should not depend on the environment,
it should just work™, even if there are no real IP addresses assigned to
the system (or container) that runs the test.

Co-authored-by: Pablo Suárez Hernández <psuarezhernandez@suse.com>
